### ISSUE TITLE
Wrong link without edit premissions

### ISSUE THREAD

**Issue by AljosaB** [silverstripe/silverstripe-lumberjack#89]
There is GridFieldSiteTreeEditButton that fixes edit link, but if current user doesn't have edit permissions the link is broken because the component is replaced with GridFieldViewButton during performReadonlyTransformation. There sould also be an extended GridFieldViewButton (GridFieldSiteTreeViewButton) that would fix the link.

Note that the permissions here are for the *parent* record. i.e. you can't edit the parent record, so the gridfield of the children is make readonly.

## PRs
- https://github.com/silverstripe/silverstripe-framework/pull/11228
- https://github.com/silverstripe/silverstripe-lumberjack/pull/90

**Comment by NightJar** [comment]
Thanks for taking the time to report this issue @AljosaB - could you please also tell us when version of SilverStripe and version of Lumberjack you're using?

**Comment by AljosaB** [comment]
SS 4.3.2 and latest Lumberjack. I created a pull request https://github.com/silverstripe/silverstripe-lumberjack/pull/90

**Comment by robbieaverill** [comment]
Are you able to provide some steps to reproduce? When I set `SiteTree::canEdit()` to false I still get the `GridFieldSiteTreeEditButton`, which matches its docblocks:

```php
 * Swaps the GridField Link out for the SiteTree edit link using {@link SiteTree::CMSEditLink()}.
...
        // No permission checks - handled through GridFieldDetailForm
        // which can make the form readonly if no edit permissions are available.
```

**Comment by AljosaB** [comment]
Just set some permissions on holder/listing page (the one with Lumberjack extension applied). For example define "who can edit this page" to administrators group. Then open the page with a user in content editors group and you'll get view buttons in lumberjack gridfield.

SS: 4.3.2
Lumberjack: master (latest)

**Comment by emteknetnz** [comment]
Linked PRs have been merged, then will be released in CMS 5.3
